node/packages/parser/src/ethereum/yearn.ts

```diff
@@ -0,0 +1,65 @@
+import { Tx } from '@shapeshiftoss/blockbook'
+import { GenericParser, TxSpecific, YearnTx } from '../types'
+import shapeShiftRouter from './abi/shapeShiftRouter'
+import yearnVault from './abi/yearnVault'
+import { ethers } from 'ethers'
+import { SHAPE_SHIFT_ROUTER_CONTRACT } from './constants'
+import { getSigHash, getYearnTokenVaultAddresses } from './utils'
+
+export class Parser implements GenericParser {
+  yearnTokenVaultAddresses: Array<string> | undefined
+
+  async parse(tx: Tx): Promise<TxSpecific<YearnTx> | undefined> {
+    const data = tx.ethereumSpecific?.data
+    if (!data) return
+
+    const abiInterface = this.getAbiInterface(data)
+    const decoded = abiInterface?.parseTransaction({ data })
+    const spender = decoded?.args._spender
```

Comment:
This would be dependent on the sigHash detected. I would prefer explicit conditional check on each possible function call check.

Reply:
Fixed up in https://github.com/shapeshift/unchained/pull/267/commits/6aef3583e6d760d005cc6034a5335760b4b5c51a